libpod/pod_status.go

```diff
@@ -6,7 +6,7 @@ import "github.com/containers/libpod/libpod/define"
 // statuses of the containers in the pod.
 // Returns a string representation of the pod status
 func (p *Pod) GetPodStatus() (string, error) {
-	ctrStatuses, err := p.Status()
+	ctrStatuses, err := p.status()
```

Comment:
This is not safe - we need to use the locked version to ensure mutual exclusion

Reply:
(The locked version, in this case, being `Status()` - the uppercased version)

Reply:
The reason for the lowercase `status` here is because `GetPodStatus` is used above in `Inspect` https://github.com/containers/libpod/pull/5320/files#diff-c0092994ef899ada726c98fafdf2e830R443, which has a lock and this was causing a double lock to happen. I guess we can have a lowercase `getPodStatus` function, wdyt @mheon?

Reply:
Given this, you cannot use it in `Inspect()` directly - you'll have to call `status()` and then `CreatePodStatusResults` from that

Reply:
I would recommend just duplicating the status code in Inspect, it's only 2 function calls

Reply:
alright, sounds good! @sujil02 ^^

Reply:
fixed.